src/main/webapp/consentGroupReview/ConsentGroupReview.js

```diff
@@ -265,14 +294,24 @@ class ConsentGroupReview extends Component {
   };
 
   handleCheck = (e) => {
+    const checked = e.target.checked;
+    const date = this.state.current.consentExtraProps.endDate;
     this.setState(prev => {
-      prev.formData.consentExtraProps.onGoingProcess = !this.state.formData.consentExtraProps.onGoingProcess;
-      prev.formData.endDate = null;
+      prev.formData.consentExtraProps.onGoingProcess = checked;
+      prev.formData.consentExtraProps.endDate = checked ? null : date;
       return prev;
-    }); // , () => this.props.updateForm(this.state.formData, "onGoingProcess"));
-    //this.props.removeErrorMessage();
+    });
   };
 
+  addDays(date, days) {
```

Comment:
General comment - there might be a better way to do this with the moment.js library. We use it in other places. Not a requirement, just a suggestion.

Reply:
I didn't thought of that, thanks Greg, will correct this in my next PR.